WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -1411,6 +1411,10 @@ - (void)savePost
     [self logSavePostStats];
 
     [self.view endEditing:YES];
+
+    //Make sure that we are uploading the latest content on the editor.
+    self.post.postTitle = self.titleText;
+    self.post.content = self.bodyText;
```

Comment:
> autoSaveContent is already being called when the image finished upload, and the race condition we see is between the save happening to CoreData and being reflected up to the self.post instance and the upload to the server.

That doesn't sound quite right.  `autosaveContent` also updates `self.post.postTitle` and `self.post.content`, just like the change in this PR.  If the race condition was `between the save happening to CoreData and being reflected up to the self.post instance and the upload` just updating `self.post.content` and `self.post.postTitle` in `savePost` shouldn't solve the race condition. 

> The autosave is done in async way

The method appears to be called syncronously on the main thread in respose to `WPEditorView webView:shouldStartLoadWithRequest:navigationType:`.  That guy is fired by an NSInvocation so maybe there is something async happening there.  But I think maybe you mean the context is saved asyncronously via `[self.post save]` (derived contexts all all that).  Regardless, I'm not seeing where core data or directly calling autosave is a factor.  We should be able to call `autosaveContent` directly without concern. 

I think this patch is right in that we just need to ensure `self.post.content` and `self.post.postTitle` are promptly updated.  However, the work is already being done in the call to `autosaveContent` so maybe it would be better to call that method immediately (since it appears to not be) rather than duplicate some of the work its already doing in another method.